src/bugreporter/odemis_bugreporter.py

```diff
@@ -0,0 +1,540 @@
+GDPR_TEXT = ("When reporting an issue, technical data from the computer will be sent " +
+    "to Delmic B.V. In addition to your name and email address, the data can " +
+    "also contain some identifiable information about your work (e.g, " +
+    "filenames of acquisitions). The sole purpose of collecting this data " +
+    "is to diagnose issues and improve the quality of the system. The data may " +
+    "be stored up to five years. The data will always be stored confidentially, " +
+    "and never be shared with any third parties or used for any commercial purposes.")
+
+
+# The next to functions will be needed to parse odemis.conf
+def _add_var_config(config, var, content):
+    """ Add one variable to the config, handling substitution
+
+    Args:
+        config: (dict) Configuration to add the found values to
+        var: (str) The name of the variable
+        content: (str) Value of the variable
+
+    Returns:
+        dict: The `config` dictionary is returned with the found values added
+
+    """
+
+    # variable substitution
+    m = re.search(r"(\$\w+)", content)
+    while m:
+        subvar = m.group(1)[1:]
+        # First try to use a already known variable, and fallback to environ
+        try:
+            subcont = config[subvar]
+        except KeyError:
+            try:
+                subcont = os.environ[subvar]
+            except KeyError:
+                logging.warning("Failed to find variable %s", subvar)
+                subcont = ""
+        # substitute (might do several at a time, but it's fine)
+        content = content.replace(m.group(1), subcont)
+        m = re.search(r"(\$\w+)", content)
+
+    logging.debug("setting %s to %s", var, content)
+    config[var] = content
+
+def parse_config(configfile):
+    """  Parse `configfile` and return a dictionary of its values
+
+    The configuration file was originally designed to be parsed as a bash script. So each line looks
+    like:
+
+        VAR=$VAR2/log
+
+    Args:
+        configfile: (str) Path to the configuration file
+
+    Returns:
+        dict str->str: Config file as name of variable -> value
+
+    """
+
+    config = DEFAULT_CONFIG.copy()
+    f = open(configfile)
+    for line in shlex.split(f, comments=True):
+        tokens = line.split("=")
+        if len(tokens) != 2:
+            logging.warning("Can't parse '%s', skipping the line", line)
+        else:
+            _add_var_config(config, tokens[0], tokens[1])
+
+    return config
+
+
+class OdemisBugreporter():
+    """
+    Class to create a bugreport. Contains functions for compressing the odemis files, opening
+    a window asking for a bugreport description, and uploading the bugreport to
+    osticket.
+    """
+    def __init__(self):
+
+        self.report_description = None  # (dict str --> str). Required fields for ticket creation.
+        self.zip_fn = None  # (str) Path to zip file.
+
+        self._executor = futures.ThreadPoolExecutor(max_workers=1)
```

Comment:
You can drop the `max_worker=1`.

Reply:
Are you sure? It seems that the ThreadPoolExecutor that I'm using needs the max_workers argument!